src/tools/__tests__/autosave.test.js

```diff
@@ -1,53 +1,46 @@
-import { updateAutosave, getAutosaveState } from '../autosave'
-import { stateFromJsmd } from '../jsmd-tools'
-import { newNotebook, newCell } from '../../editor-state-prototypes'
+/* eslint-disable no-unused-vars */
+import { IDBFactory, IDBKeyRange, reset } from 'shelving-mock-indexeddb';
 
-let states
+window.indexedDB = new IDBFactory();
+window.IDBKeyRange = IDBKeyRange;
 
-describe('updateAutosave', () => {
+describe('it test the autosave', () => {
   beforeEach(() => {
-    const originalState = newNotebook()
-    updateAutosave(originalState, true)
-
-    // add some state
-    const stateUpdate = {
-      cells: [newCell(1, 'code'), newCell(2, 'markdown')],
-      title: 'autosaved title',
-    }
-    stateUpdate.cells[0].selected = true
-    const updatedState = Object.assign({}, originalState, stateUpdate)
-
-    states = {
-      originalState,
-      updatedState,
-    }
-  })
-
-  it('saves over original when asked to', () => {
-    updateAutosave(states.updatedState, true)
-    const newAutosavedState = getAutosaveState(states.updatedState)
-    expect(Object.keys(newAutosavedState).sort()).toEqual(['originalCopy', 'originalSaved'])
-
-    const originalCopyState = stateFromJsmd(newAutosavedState.originalCopy)
-    expect(originalCopyState.cells).toEqual(states.updatedState.cells)
-    expect(originalCopyState.title).toEqual(states.updatedState.title)
-  })
-
-  it('only updates dirty copy when not asked to write over original', () => {
-    updateAutosave(states.updatedState, false)
-    const newAutosavedState = getAutosaveState(states.updatedState)
-    expect(Object.keys(newAutosavedState).sort())
-      .toEqual(['dirtyCopy', 'dirtySaved', 'originalCopy', 'originalSaved'])
-
-    const originalCopyState = stateFromJsmd(newAutosavedState.originalCopy)
-    expect(originalCopyState.cells).toEqual(states.originalState.cells)
-    expect(originalCopyState.title).toEqual(states.originalState.title)
-
-    const dirtyCopyState = stateFromJsmd(newAutosavedState.dirtyCopy)
-    expect(dirtyCopyState.cells).toEqual(states.updatedState.cells)
-    expect(dirtyCopyState.title).toEqual(states.updatedState.title)
-  })
-})
-
-describe('updateAutosave', () => {
+    reset();
```

Comment:
With these changes, This unit test isn't testing the actual codepaths we're using in the client, so I don't think we can use it. Is there not a way to get dexie to use the mock indexdb? I feel like there should be, somehow